### PR TITLE
feat(mime): add mime/content_type module

### PR DESCRIPTION
**Pull request by kt3k**
This PR adds `contentType` method in `std/mime`. The design is inspired by `TypeByExtension` function from Go's mime package

refs:
- Go's mime package https://pkg.go.dev/mime
- past discussion about previous `std/media_types` module https://github.com/denoland/deno/issues/4292

### CODE REVIEW

File: mime/content_type.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+const EXTRACT_TYPE_REGEXP = /^\s*([^;\s]*)(?:;|\s|$)/;
+const TEXT_TYPE_REGEXP = /^text\//i;
+
+const extensionToType = new Map<string, string>();
+
+for (const [name, type] of Object.entries(db)) {
+  if ("extensions" in type) {
+    let mimeType = name;
+    const cs = charset(mimeType, "charset" in type ? type.charset : undefined);
```
Comment (cjihrig) [inline mime/content_type.ts:17]: Can `"charset" in type ? type.charset : undefined` be replaced with just `type.charset`, or does TypeScript complain?

**Review by cjihrig — APPROVED** [review]

File: mime/content_type.ts (hunk 1 above)
Comment (kt3k) [reply mime/content_type.ts:17]: Yup, typescript complain about it. It's typed as union of various shaped objects. Some have `charset` and some don't.

スクリーンショット 2022-05-30 19 08 18

**Comment by bartlomieju** [comment]
Looks like this PR conflicts with https://github.com/denoland/deno_std/pull/2286

**Comment by lucacasonato** [comment]
> Looks like this PR conflicts with https://github.com/denoland/deno_std/pull/2286

I would rather say #2286 conflicts with this PR, as this one came first :)

**Comment by kitsonk** [comment]
> > Looks like this PR conflicts with #2286
> 
> I would rather say #2286 conflicts with this PR, as this one came first :)

Though I suggested in multiple channels, as maintained of `media_types` that I would work to re-integrate it back into `std`. 😕

**Comment by kt3k** [comment]
closing in favor of #2286. #2286 is (roughly) a superset of this. So we don't need this anymore.

**Pull request closed without merging**